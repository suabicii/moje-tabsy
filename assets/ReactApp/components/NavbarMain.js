import React from "react";

function NavbarMain() {
    return (
        //Navbar -->
        <nav id="main-navbar" className="navbar navbar-expand-lg navbar-dark bg-primary fixed-top">
            {/*Container wrapper*/}
            <div className="container-fluid">
                {/*// Toggle button -->*/}
                <button
                    className="navbar-toggler"
                    type="button"
                    data-bs-toggle="collapse"
                    data-bs-target="#sidebarMenu"
                    aria-controls="sidebarMenu"
                    aria-expanded="false"
                    aria-label="Toggle navigation"
                >
                    <i className="fas fa-bars"></i>
                </button>

                {/*// Brand -->*/}
                <a className="navbar-brand fst-italic d-none d-lg-block" href="#">Moje-Tabsy.pl <i
                    className="fa-solid fa-pills"></i></a>
                {/*// Search form -->*/}
                <form className="d-none d-md-flex input-group w-auto my-auto">
                    <input
                        id="searchInput"
                        autoComplete="off"
                        type="search"
                        className="form-control rounded-start dashboard__search"
                        placeholder='Wyszukaj'
                    />
                    <label htmlFor="searchInput" className="visually-hidden">Wyszukaj</label>
                    <span className="input-group-text border-0"><i className="fas fa-search"></i></span>
                </form>

                {/*// Right links -->*/}
                <ul className="navbar-nav ms-auto d-flex flex-row">
                    {/*// Notification dropdown -->*/}
                    <li className="nav-item dropdown">
                        <a
                            className="nav-link me-3 me-lg-0 notifications-dropdown dropdown-toggle hidden-arrow"
                            href="#"
                            id="notificationDropdown"
                            role="button"
                            data-bs-toggle="dropdown"
                            aria-expanded="false"
                        >
                            <i className="fas fa-bell"></i>
                            <span className="badge rounded-pill badge-notification bg-danger">1</span>
                        </a>
                        <ul
                            className="dropdown-menu dropdown-menu-end"
                            aria-labelledby="notificationDropdown"
                        >
                            <li>
                                <a className="dropdown-item" href="#">Some news</a>
                            </li>
                            <li>
                                <a className="dropdown-item" href="#">Another news</a>
                            </li>
                            <li>
                                <a className="dropdown-item" href="#">Something else here</a>
                            </li>
                        </ul>
                    </li>

                    {/*// Icon dropdown -->*/}
                    <li className="nav-item dropdown">
                        <a
                            className="nav-link me-3 me-lg-0 dropdown-toggle hidden-arrow"
                            href="#"
                            id="navbarDropdown"
                            role="button"
                            data-bs-toggle="dropdown"
                            aria-expanded="false"
                        >
                            <i className="m-0">ðŸ‡µðŸ‡±</i>
                        </a>
                        <ul className="dropdown-menu dropdown-menu-end" aria-labelledby="navbarDropdown">
                            <li>
                                <a className="dropdown-item" href="#"
                                ><i>ðŸ‡µðŸ‡±</i> Polski
                                    <i className="fa fa-check text-success ms-2"></i
                                    ></a>
                            </li>
                            <li>
                                <hr className="dropdown-divider"/>
                            </li>
                            <li>
                                <a className="dropdown-item" href="#"><i>ðŸ‡¬ðŸ‡§</i> English</a>
                            </li>
                        </ul>
                    </li>

                    {/*// Avatar -->*/}
                    <li className="nav-item dropdown">
                        <a
                            className="nav-link dropdown-toggle hidden-arrow d-flex align-items-center"
                            href="#"
                            id="navbarDropdownMenuLink"
                            role="button"
                            data-bs-toggle="dropdown"
                            aria-expanded="false"
                        >
                            <i className="fa-solid fa-circle-user fs-4"></i>
                        </a>
                        <ul
                            className="dropdown-menu dropdown-menu-end"
                            aria-labelledby="navbarDropdownMenuLink"
                        >
                            <li>
                                <a className="dropdown-item" href="#">MÃ³j profil</a>
                            </li>
                            <li>
                                <a className="dropdown-item" href="#">Ustawienia</a>
                            </li>
                            <li>
                                <a className="dropdown-item" href="/logout">Wyloguj siÄ™</a>
                            </li>
                        </ul>
                    </li>
                </ul>
            </div>
            {/*// Container wrapper -->*/}
        </nav>
        //Navbar -->
    );
}

export default NavbarMain;