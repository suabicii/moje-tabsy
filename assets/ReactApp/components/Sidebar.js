import React from "react";
import {NavLink} from "react-router-dom";
import {mainRoute} from "../routers/AppRouter";

function Sidebar() {
    return (
        // Sidebar -->
        <nav id="sidebarMenu" className="collapse d-lg-block sidebar collapse bg-white">
            <div className="position-sticky">
                <div className="list-group list-group-flush mx-3 mt-4">
                    <NavLink to={mainRoute} className="list-group-item list-group-item-action py-2 ripple" end>
                        <i className="fas fa-tachometer-alt fa-fw me-3"></i><span>Strona główna</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/drug-list`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-jar fa-fw me-3"></i><span>Moje leki i suplementy</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/profile`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-user fa-fw me-3"></i><span>Mój profil</span>
                    </NavLink>
                    <NavLink to={`${mainRoute}/settings`} className="list-group-item list-group-item-action py-2 ripple">
                        <i className="fa-solid fa-gear fa-fw me-3"></i><span>Ustawienia</span>
                    </NavLink>
                    <a href="/logout" className="list-group-item list-group-item-action py-2 ripple"
                    ><i className="fa-solid fa-power-off fa-fw me-3"></i><span>Wyloguj się</span></a
                    >
                </div>
            </div>
        </nav>
        //Sidebar -->
    );
}

export default Sidebar;