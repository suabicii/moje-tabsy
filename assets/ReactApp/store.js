import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});