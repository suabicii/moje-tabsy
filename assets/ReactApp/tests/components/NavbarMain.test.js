/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import NavbarMain from "../../components/NavbarMain";
import {Provider} from "react-redux";
import store from "../../store";
import {fireEvent, render, screen} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import Sidebar from "../../components/Sidebar";

function WrappedComponent() {
    return (
        <Provider store={store}>
            <BrowserRouter>
                <NavbarMain/>
                <Sidebar/>
            </BrowserRouter>
        </Provider>
    );
}

const changeTheme = require('../../utils/changeTheme');
let changeThemeSpy;


beforeAll(() => {
    changeThemeSpy = jest.spyOn(changeTheme, 'default');
});

afterAll(() => {
    jest.clearAllMocks();
});

it('should correctly render main navbar', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Provider store={store}>
            <NavbarMain/>
        </Provider>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should change theme to light', () => {
    render(<WrappedComponent/>);

    expect(changeThemeSpy).toHaveBeenLastCalledWith(false);
});

it('should change theme to dark after clicking toggler', () => {
    render(<WrappedComponent/>);

    fireEvent.click(screen.getByTestId('dark-theme-toggle'));

    expect(changeThemeSpy).toBeCalledWith(true);
});
