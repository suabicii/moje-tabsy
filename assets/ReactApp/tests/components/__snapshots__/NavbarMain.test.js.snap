// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render main navbar 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
  id="main-navbar"
>
  <div
    className="container-fluid"
  >
    <button
      aria-controls="sidebarMenu"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#sidebarMenu"
      data-bs-toggle="collapse"
      type="button"
    >
      <i
        className="fas fa-bars"
      />
    </button>
    <a
      className="navbar-brand fst-italic d-none d-lg-block"
      href="#"
    >
      MediMinder 
      <i
        className="fa-solid fa-pills"
      />
    </a>
    <ul
      className="navbar-nav ms-auto d-flex flex-row"
    >
      <li
        className="nav-item dropdown avatar"
      >
        <a
          aria-expanded="false"
          className="nav-link dropdown-toggle hidden-arrow d-flex align-items-center"
          data-bs-toggle="dropdown"
          href="#"
          id="navbarDropdownMenuLink"
          role="button"
        >
          <i
            className="fa-solid fa-circle-user fs-4"
          />
        </a>
        <ul
          aria-labelledby="navbarDropdownMenuLink"
          className="dropdown-menu dropdown-menu-end position-absolute"
        >
          <li>
            <Link
              className="dropdown-item"
              to="/dashboard/profile"
            >
              Mój profil
            </Link>
          </li>
          <li>
            <Link
              className="dropdown-item"
              to="/dashboard/settings"
            >
              Ustawienia
            </Link>
          </li>
          <li>
            <a
              className="dropdown-item"
              href="/logout"
            >
              Wyloguj się
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</nav>
`;
